src/contrib/mlir/compiler/pass/mlir_subgraph_extraction.cpp

```diff
@@ -384,6 +386,45 @@ void MLIRSubgraphExtractionPass::sanity_check(std::shared_ptr<Function> func, No
     }
 }
 
+void MLIRSubgraphExtractionPass::encapsulate_nodes(NodeVector& ck_nodes)
```

Comment:
Again, this should be part of CK construction logic and not part of the pass. In case we need to construct CKs elsewhere.